cmd/admin-handlers.go

```diff
@@ -80,3 +80,77 @@ func (adminAPI adminAPIHandlers) ServiceRestartHandler(w http.ResponseWriter, r
 	w.WriteHeader(http.StatusOK)
 	sendServiceCmd(globalAdminPeers, serviceRestart)
 }
+
+// ListLocksHandler - GET /?lock&bucket=mybucket&prefix=myprefix&older-than=rel_time
+// HTTP header x-minio-operation: list
+// ---------
+// Lists locks held on a given bucket, prefix and relative time.
+func (adminAPI adminAPIHandlers) ListLocksHandler(w http.ResponseWriter, r *http.Request) {
+	adminAPIErr := checkRequestAuthType(r, "", "", "")
+	if adminAPIErr != ErrNone {
+		writeErrorResponse(w, r, adminAPIErr, r.URL.Path)
+		return
+	}
+	vars := mux.Vars(r)
+	bucket := vars["bucket"]
+	prefix := vars["prefix"]
+	relTimeStr := vars["older-than"]
+
+	if !IsValidBucketName(bucket) {
+		writeErrorResponse(w, r, ErrInvalidBucketName, r.URL.Path)
+		return
+	}
+	if !IsValidObjectPrefix(prefix) {
+		writeErrorResponse(w, r, ErrInvalidObjectName, r.URL.Path)
+	}
+	relTime, err := time.ParseDuration(relTimeStr)
+	if err != nil {
+		writeErrorResponse(w, r, ErrInvalidDuration, r.URL.Path)
```

Comment:
The error message is returned as part of error response. How does it help to log it on server-side too?